### ISSUE TITLE
Every container or host system reboot resets Windows network settings

### ISSUE THREAD

**Issue by itiligent** [winapps-org/winapps#161]
Every time the Windows docker container is either rebooted, or the host Linux system is restarted, the Windows network settings are not carried over at next boot requiring manual intervention to re-establish network groups and sharing. 

Pictured is after the 4th reboot of my test container that built with the default compose file. Each reboot requires I log in interactively to reset this. Doe's anyone know what am I missing?

![image](https://github.com/user-attachments/assets/525eb478-d0f3-4824-935f-59723242b793)

**Comment by KernelGhost** [comment]
Hello, thank you for your interest in the project.

It seems that your question pertains to the behaviour of the Docker container itself, which would be best addressed by the [dockur/windows team](https://github.com/dockur/windows/issues). If you decide to open an issue on their repository, please share the link here for the benefit of other users.

**Comment by gordoncheong** [comment]
Maybe [this](https://superuser.com/questions/1214281/how-to-hide-or-disable-allow-pc-to-be-discoverable-on-this-network) can help you?

**Comment by itiligent** [comment]
For those still looking for a workaround:

1. Stop the annoying prompt by adding this key: 

`[HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Network\NewNetworkWindowOff]`

2. Set this script to run as a task at boot to keep the network profiles tidy (you will need to adapt it if you have multiple virutal NICs. It will remove any unused profiels and set the current network adatper name to "WinApps"
```
# Get the current network profile name
$currentProfile = (Get-NetConnectionProfile).Name

# Get all profiles from the registry
$profilesKey = "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\NetworkList\Profiles"
$profiles = Get-ChildItem -Path $profilesKey

foreach ($profile in $profiles) {
    $profilePath = "$profilesKey\$($profile.PSChildName)"
    $profileName = (Get-ItemProperty -Path $profilePath).ProfileName

    # Remove profiles that don't match the current one
    if ($profileName -ne $currentProfile) {
        Remove-Item -Path $profilePath -Recurse
        Write-Host "Deleted profile: $profileName"
    }
}

# Change the current profile name to "WinApps"
$profiles = Get-ChildItem -Path $profilesKey
foreach ($profile in $profiles) {
    $profilePath = "$profilesKey\$($profile.PSChildName)"
    $profileName = (Get-ItemProperty -Path $profilePath).ProfileName

    if ($profileName -eq $currentProfile) {
        # Update the profile name
        Set-ItemProperty -Path $profilePath -Name "ProfileName" -Value "WinApps"
        Write-Host "Renamed profile to: WinApps"
    }
}
```

**Comment by oskardotglobal** [comment]
@itiligent Can we set this script up as a boot task from inside installer.bat?

**Comment by itiligent** [comment]
Yes, I was already looking at that yesterday. As a batch file it's not really workable, but it's possible to create the powershell script and call it as a scheduled task at boot all via the install.bat. I've built a time sync function as a scheduled task in install.bat already and it works OK. My install script only needs a little more adaptation for fedora but I've been short of time lately

**Comment by itiligent** [comment]
FYI I just put this idea into a PR

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by itiligent** [comment]
Solved in PR #259
